cypress/e2e/permit2.test.ts

```diff
@@ -5,212 +5,187 @@ import { getTestSelector } from '../utils'
+  // The same tokens are used for all permit2 tests.
   const INPUT_TOKEN = DAI
   const OUTPUT_TOKEN = USDC_MAINNET
-  const TEST_BALANCE_INCREMENT = 0.01
 
   beforeEach(() => {
     // Sets up a swap between INPUT_TOKEN and OUTPUT_TOKEN.
     cy.visit(`/swap/?inputCurrency=${INPUT_TOKEN.address}&outputCurrency=${OUTPUT_TOKEN.address}`, {
       ethereum: 'hardhat',
     })
-    cy.get('#swap-currency-input .token-amount-input').type(TEST_BALANCE_INCREMENT.toString())
+    cy.get('#swap-currency-input .token-amount-input').type('0.01')
   })
 
   /** Asserts permit2 has a max approval for spend of the input token on-chain. */
   function expectTokenAllowanceForPermit2ToBeMax() {
     // check token approval
-    return cy
-      .hardhat()
+    cy.hardhat()
       .then(({ approval, wallet }) => approval.getTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN }))
       .should('deep.equal', MaxUint256)
   }
 
   /** Asserts the universal router has a max permit2 approval for spend of the input token on-chain. */
-  function expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime: number) {
-    return cy
-      .hardhat()
+  function expectPermit2AllowanceForUniversalRouterToBeMax() {
+    cy.hardhat()
       .then((hardhat) => hardhat.approval.getPermit2Allowance({ owner: hardhat.wallet, token: INPUT_TOKEN }))
       .then((allowance) => {
         cy.wrap(MaxUint160.eq(allowance.amount)).should('eq', true)
         // Asserts that the on-chain expiration is in 30 days, within a tolerance of 40 seconds.
-        const expected = Math.floor((approvalTime + 2_592_000_000) / 1000)
+        const THIRTY_DAYS_SECONDS = 2_592_000
+        const expected = Math.floor(Date.now() / 1000 + THIRTY_DAYS_SECONDS)
         cy.wrap(allowance.expiration).should('be.closeTo', expected, 40)
       })
   }
 
   it('swaps when user has already approved token and permit2', () => {
-    cy.hardhat().then(({ approval, wallet }) => {
-      approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN })
-      approval.setPermit2Allowance({ owner: wallet, token: INPUT_TOKEN })
-    })
+    cy.hardhat().then(({ approval, wallet }) =>
+      Promise.all([
+        approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN }),
+        approval.setPermit2Allowance({ owner: wallet, token: INPUT_TOKEN }),
+      ])
+    )
     initiateSwap()
-    cy.get(getTestSelector('confirmation-close-icon')).click()
-    // Verifies that there is a successful swap notification.
-    cy.contains('Swapped').should('exist')
+
+    // Verify transaction
+    cy.contains('Success')
+    cy.get(getTestSelector('popups')).contains('Swapped')
   })
 
   it('swaps after completing full permit2 approval process', () => {
-    cy.hardhat().then(({ provider }) => {
-      cy.spy(provider, 'send').as('permitApprovalSpy')
-    })
+    cy.hardhat().then((hardhat) => hardhat.setAutomine(false))
```

Comment:
This appears to be flaky - sometimes the subsequent commands will get in before the command to toggle automine. See https://cloud.cypress.io/projects/yp82ef/runs/11453/overview/bcb08413-dfde-4d9d-8ddd-5878a4e3d021?roarHideRunsWithDiffGroupsAndTags=1.

Reply:
I've tried to refactor it to what I _hope_ will be less flaky, but I'll also investigate how cypress-hardhat works to see if it's misconfigured.